### DISCUSSION TITLE
Cutting edge deep learning

### DISCUSSION THREAD

**Discussion by edvardoss** [IyarLin/R-advantages-over-python#6] [General]
> For the most cutting edge models it might be preferable to use python directly (instead of through reticulate)

I think that using deep neural networks now in R is not so difficult as in Python. Of course, you can find difficulties in some specific tasks (for example, in the YOLO object recognition), but on the whole it is not at all such a big abyss as it was before.
For example, Torch for R came out without dependence on Python
[Please allow me to introduce myself: Torch for R](https://medium.com/pytorch/please-allow-me-to-introduce-myself-torch-for-r-9ea0f361ea7e)
Also for working with deep neural networks there is a wonderful H2O package.
/*Off Topic*/
It's sad to realize in Moscow (a huge megalopolis with a giant number of employers) is now almost impossible to find a job on R, Python is required everywhere.
Therefore, squeezing the teeth from anger on an uncomfortable Pandas, NamPy, Matplotlib and SciKitLearn I slowly learn Python every time heat recalling Tidyverse, data.table, Tidymodels, Shiny.

**Comment by IyarLin** [comment]
I feel your pain. The only reason I conceded python **might** be preferable for some **really cutting edge DL** is because I'm not that familiar with DL and decided to give python some points too. Here in Israel the situation is very similar with the vast majority of employers asking for python (though experience with R usually counts). 

I hope this repo and accompanying TDS articles will make some change for the better.
